.github/workflows/main.yml

```diff
@@ -11,103 +11,38 @@ jobs:
   enforce-g_malloc:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-      with:
-        # Clone the whole branch, we have to fetch tags later
-        fetch-depth: 0
-
+    - uses: actions/checkout@v4
     - name: enforce-g_malloc
       run: "! git grep -P '(?<!g_)malloc' ':!.valgrind.suppressions' ':!.github/workflows/main.yml'"
 
   build:
-    strategy:
-      matrix:
-        CC:
-          - clang
-          - gcc
-        distro:
```

Comment:
Does it not run on all distro's anymore when moving to meson?

Reply:
No, im just simplifying the CI.

Reply:
Additionally can be somewhat solved with #1232

Reply:
It's actually useful to run the CI on multiple distro's because of their slight differences. This goes unnoticed in writing the code, but will get caught by the CI. For example, using a GLib feature that was only added in a newer version didn't compile in an old CI image. Now a check could be added for glib version and it compiles on all distro's again.

Reply:
This is why a version requirement can be added into the Meson glib dependency.

Reply:
Still, testing on multiple distro's provides value in my opinion (for example alpine has musl libc, which has some differences from gnu libc). So please revert this change